polynote-frontend/polynote/ui/component/messaging/dispatcher.ts

```diff
@@ -0,0 +1,580 @@
+/**
+ * The Dispatcher is used to handle actions initiated by the UI.
+ * It knows whether an Action should be translated to a message and then sent on the socket, or if it should be
+ * handled by something else.
+ */
```

Comment:
The Dispatchers are how UI events get propagated back to the server and/or are reflected in the State. 

The UI components themselves should never update the state, only the appropriate dispatcher (for changes generated internally) or receiver (for changes generated externally) should update the state. So, the components send events to the dispatcher in order to notify it of events.